docs/how-to/use_prompting.md

```diff
@@ -0,0 +1,210 @@
+
+Let's use a RAG example as a case study:
+
+```python
+from pydantic import BaseModel
+
+from ragbits.core.prompt import Prompt
+
+
+class QueryWithContext(BaseModel):
+    """
+    Input format for the QueryWithContext.
+    """
+
+    query: str
+    context: list[str]
+
+
+class RAGPrompt(Prompt[QueryWithContext]):
+    """
+    A simple prompt for RAG system.
+    """
+
+    system_prompt = """
+    You are a helpful assistant. Answer the QUESTION that will be provided using CONTEXT.
+    If in the given CONTEXT there is not enough information refuse to answer.
+    """
+
+    user_prompt = """
+    QUESTION:
+    {{ query }}
+
+    CONTEXT:
+    {% for item in context %}
+        {{ item }}
+    {% endfor %}
+    """
+
+...
+
+prompt = RAGPrompt(QueryWithContext(query=message, context=[i.get_text_representation() for i in results]))
+response = await self._llm.generate(prompt)
+print(response)
+...
+```
+### How to configure `Prompt`'s output data type
+#### Defining output as a Pydantic Model
+You can define the output of a prompt as a Pydantic model by specifying the output type as a generic parameter.
+```python
+from pydantic import BaseModel
+
+from ragbits.core.prompt import Prompt
+
+
+class QueryWithContext(BaseModel):
+    """
+    Input format for the QueryWithContext.
+    """
+
+    query: str
+    context: list[str]
+
+
+class ChatAnswer(BaseModel):
+    """
+    Output format for the ChatAnswer.
+    """
+
+    answer: str
+
+
+class RAGPrompt(Prompt[QueryWithContext, ChatAnswer]):
+    """
+    A simple prompt for RAG system.
+    """
+
+    system_prompt = """
+    You are a helpful assistant. Answer the QUESTION that will be provided using CONTEXT.
+    If in the given CONTEXT there is not enough information refuse to answer.
+    """
+
+    user_prompt = """
+    QUESTION:
+    {{ query }}
+
+    CONTEXT:
+    {% for item in context %}
+        {{ item }}
+    {% endfor %}
+    """
+
+...
+
+prompt = RAGPrompt(QueryWithContext(query=message, context=[i.get_text_representation() for i in results]))
+response = await self._llm.generate(prompt)
+print(response.answer)
+...
+```
+#### Configuring output as a simple type
+You can configure the ouput as a simple type, such as `bool`
+```python
+from pydantic import BaseModel
+
+from ragbits.core.prompt import Prompt
+
+
+class RoleInput(BaseModel):
+    role: str
+
+
+class BooleanPrompt(Prompt[RoleInput, bool]):
+    user_prompt = "Are you {{ role }}? Answer 'yes' or 'no' only."
+
+
+boolean_prompt = BooleanPrompt(RoleInput(role = "an AI Assistant"))
+assert boolean_prompt.parse_response("true") is True
+```
+Please note that an actual response from LLM is going to raise `ResponseParsingError`. This is because the response is not natively parsable to boolean:
+```python
+llm = LiteLLM("gpt-4o-2024-08-06", use_structured_output=True)
```

Comment:
i don't think that we should include non-working code here - only the correct example with description why do we need to implement parsing should be included.